packages/helix/build.sh

```diff
@@ -0,0 +1,50 @@
+# http(s) link to package home page.
+TERMUX_PKG_HOMEPAGE="https://helix-editor.com/"
+
+# One-line, short package description.
+TERMUX_PKG_DESCRIPTION="A post-modern modal text editor written in rust"
+
+# License.
+TERMUX_PKG_LICENSE="MPL-2.0"
+
+# Who cares about package.
+# Specify yourself (Github nick, or name + email) if you wish to maintain the
+# package, fix its bugs, etc. Otherwise specify "@termux".
+# Please note that unofficial repositories are not allowed to reference @termux
+# as their maintainer.
+TERMUX_PKG_MAINTAINER="@termux"
+
+# Version.
+TERMUX_PKG_VERSION=0.2.1
+
+# URL to archive with source code
+#TERMUX_PKG_SRCURL=https://github.com/helix-editor/helix/archive/v${TERMUX_PKG_VERSION}.tar.gz
+
+# SHA-256 checksum of the source code archive.
+#TERMUX_PKG_SHA256=ba388eded1246f00b5287ad9cfdaeebf24eada42729bf61b936bb70488cfb4fd
+TERMUX_PKG_SKIP_SRC_EXTRACT=true
+TERMUX_PKG_BUILD_IN_SRC=true
+
+termux_step_get_source(){
+  mkdir -p "$TERMUX_PKG_SRCDIR" && cd "$TERMUX_PKG_SRCDIR"
+  git clone --recurse-submodules --shallow-submodules -j8 https://github.com/helix-editor/helix --branch=v0.2.1 # this is tag, actually! 
```

Comment:
This is inappropriate way of getting source code. Please look the ./packages directory of examples for both tarball and git methods.

Reply:
If default methods of getting source code that are defined in build-package.sh (./scripts/build/*) are not suitable, you still can go with your own. But please put the checkout into cache dir (`TERMUX_PKG_CACHEDIR`) and then copy it to source.

Reply:
Source repo neither has any branch related to specific release nor do their archive has all the submodules included. So, I can not think of any other way of getting source, but if you could suggest something.....

Reply:
use this...
```
$TERMUX_PKG_SRCURL=https://github.com/helix-editor/helix.git
```
The script will automatically include submodules and also checkout with the correct tag if the version is right ....

Reply:
Thank you!

Reply:
@xeffyr  updated